src/react/providers/chat-room-provider.tsx

```diff
@@ -77,42 +128,99 @@ export const ChatRoomProvider: React.FC<ChatRoomProviderProps> = ({
 }) => {
   const client = useChatClient();
   const logger = useLogger();
-  logger.trace(`ChatRoomProvider();`, { roomId, options, release, attach });
+  logger.debug(`ChatRoomProvider();`, { roomId, options, release, attach });
+
+  // Set the initial room promise, we do this in a function to avoid rooms.get being called
+  // every time the component re-renders
+  // In StrictMode this will be called twice one after the other, but that's ok
+  const [value, setValue] = useState<ChatRoomContextType>(() => {
+    logger.debug(`ChatRoomProvider(); initializing value`, { roomId, options });
+    return { room: client.rooms.get(roomId, options).catch(), roomId: roomId, options: options, client: client };
```

Comment:
What's the purpose of this `catch()`?

If we want to allow the error from rooms.get to propagate, we remove the catch().

If we want to suppress the error, we need to pass an empty function. But I'm guessing we don't want this promise of type `Promise<Room>` to resolve to `undefined`, that would be breaking the interface.

Reply:
Seems to work, but I've made it a bit better for us :)